packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -4186,8 +4159,15 @@ export function writeResourcesForBoundary(
   currentlyRenderingBoundaryHasStylesToHoist = false;
   destinationHasCapacity = true;
 
-  // Flush each Boundary resource
-  boundaryResources.forEach(flushStyleTagsLateForBoundary, destination);
+  // Flush style tags for each precedence this boundary depends on
+  boundaryResources.precedences.forEach(
+    flushStyleTagsLateForBoundary,
+    destination,
+  );
+
+  // Determine if this boundary has stylesheets that need to be awaited upon completion
+  boundaryResources.sheets.forEach(hasStylesToHoist);
```

Comment:
would be nice if `Set.prototype.some` existed. Would love to bail out early but not sure if using a raw iterator is slower enough in the common case (few sheets) to be worth it